include/threads.php

```diff
@@ -97,15 +97,26 @@ function add_shadow_thread($itemid) {
 			unset($item[0]['id']);
 			$item[0]['uid'] = 0;
 			$item[0]['origin'] = 0;
+			$item[0]['wall'] = 0;
 			$item[0]['contact-id'] = get_contact($item[0]['author-link'], 0);
+
+			if (in_array($item[0]['type'], array("net-comment", "wall-comment"))) {
+				$item[0]['type'] = 'remote-comment';
+			} elseif ($item[0]['type'] == 'wall') {
+				$item[0]['type'] = 'remote';
+			}
+
 			$public_shadow = item_store($item[0], false, false, true);
 
 			logger("Stored public shadow for thread ".$itemid." under id ".$public_shadow, LOGGER_DEBUG);
 		}
 	}
 }
 
-function add_shadow_entry($item) {
+function add_shadow_entry($itemid) {
```

Comment:
I don't know what the function does. I'm always writing code without thinking ;-)